### ISSUE TITLE
Activate HTTPS by Default?

### ISSUE THREAD

**Issue by rei-vilo** [mozilla-iot/gateway#688]
For security reasons, shouldn't HTTPS be activated by default on the gateway?

<img width="236" alt="capture 2018-02-18 a 12 11 59" src="https://user-images.githubusercontent.com/951776/36351117-fed8cbf0-14a4-11e8-808b-d07ee3c6cf6a.png">

**Comment by dhylands** [comment]
It is. Once you get past the initial setup (which is done via http) and the domain is registered, then the gateway runs in https mode and has a redirector to redirect http requests to https.

It looks like you're using a .local address, which won't match with any of the certificates that we create (which are all custom-name.mozilla-iot.org), and this mismatch is why you're getting the lock with the slash through it. You can check your registered domain through settngs (IIRC).

**Comment by rei-vilo** [comment]
Using the IP address instead of the `.local` address displays the same warning and the page is blank with "The connection has failed."

<img width="259" alt="capture 2018-02-18 a 17 53 15" src="https://user-images.githubusercontent.com/951776/36354444-de1f3a86-14d4-11e8-805a-0a2755ab1293.png">

Does that mean I need to configure the gateway from scratch again?

**Comment by rei-vilo** [comment]
Ok, so I started again from scratch.

As you suggested, I connected the browser to the IP address and not the `.local` address and was able to get HTTPS.

However, just after page **Connect to a WiFi network?**, the installation procedure suggests to connect to `gateway.local` instead of the IP address.

<img width="400" alt="capture 2018-02-18 a 18 53 26" src="https://user-images.githubusercontent.com/951776/36355085-f2b87c4c-14dd-11e8-9200-71d379997374.png">

Maybe rewording the installation procedure would make that specific point more clear.

**Comment by benfrancis** [comment]
I don't think Dave was suggesting you use the local IP. HTTPS doesn't work on local networks, except if you generate a self signed certificate and accept a security exception in your browser.

HTTPS will work if you access the gateway over the *.mozilla-iot.org subdomain as intended, or if you configure your own domain name and SSL certificates.

Access via a local IP address or .local domain will not give you a green lock icon. This is a limitation of how HTTPS works because it relies on the Internet domain name system. The [Local HTTPS Community Group](https://www.w3.org/community/httpslocal/) is looking for solutions to this problem.

**Comment by benfrancis** [comment]
(See the README for how to generate a self-signed certificate https://github.com/mozilla-iot/gateway#download-and-build-gateway but you will still need to accept a security exception in your browser).

**Comment by rei-vilo** [comment]
Done, thank you!

I can now connect through the `.mozilla-iot.org` subdomain and local IP, both with HTTPS.

Connecting the very first time with the IP address and not the `gateway.local` made it possible, hence my suggestion.
